test_aodntools/ncwriter/test_template.py

```diff
@@ -348,12 +348,16 @@ def setUp(self):
                 'X': {
                     '_dimensions': ['TIME'],
                     '_datatype': 'float32',
+                    'valid_min': 1,
+                    'valid_max': 5,
                     '_FillValue': -999.,
                     '_data': self.data_array
                 },
                 'Y': {
                     '_dimensions': ['TIME'],
                     '_datatype': 'float32',
+                    'valid_min': -4,
+                    'valid_max': 5,
```

Comment:
You're testing the exact same case 4 times (variable is `float32`, attribute looks like integer). How about changing one of the variables to `float64`, and maybe adding an `int32`, and specifying some of these numbers with a decimal point.

Should also test the correct handling of 
* 'valid_range', as it is array-valued (e.g. `'valid_range': [-4, 5]`), and
* '_FillValue', which is a special attribute, handled by the NetCDF4 library.